### PR TITLE
document new-app pipeline strategy and some cleanups

### PR DESCRIPTION
**Pull request by jim-minter**
https://trello.com/c/H6ZwrpCo/995-5-new-app-new-build-support-for-pipeline-buildconfigs-techdebt

Documentation for the above feature - targeted for 1.5 / 3.5.

### CODE REVIEW

**Comment by ahardin-rh** [comment]
Thanks!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/new_app.adoc#dev-guide-new-app[Creating New Applications]
|Added information about the `pipeline` build strategy.
%

**Comment by adellape** [comment]
Chatted w/ @ahardin-rh offline. This one got weird when I was doing the publishing for the downstream branches this week because I didn't realize it was part of my https://github.com/openshift/openshift-docs/pull/3275 ALM restructure (merged within a few days of each other, and moved the `new_app.adoc` file down a level), and I didn't realize when publishing that this PR mixed some 3.5-only stuff with normal edits.

I have since applied all the normal cleanup edits (but **not** the pipeline stuff) to the downstream branches (enterprise-3.3, dedicated, and online), and on master I've broken out the 3.5-only pipeline parts into a separate PR: removed via https://github.com/openshift/openshift-docs/pull/3396 and added via https://github.com/openshift/openshift-docs/pull/3397. That way we can more easily manage only publishing the pipeline stuff when we update 3.5 docs (via https://github.com/openshift/openshift-docs/pull/3397), but master (Origin docs) still gets it now.

TL;DR - I'm removing labels+milestone from this PR since it's been taken care of so to speak, and now the only outstanding piece is cherry-picking https://github.com/openshift/openshift-docs/pull/3397 for the 3.5 doc update.
